test/core/data.wast

```diff
@@ -333,3 +333,37 @@
 ;;   (module (memory 1) (data (global.get $g)) (global $g (mut i32) (i32.const 0)))
 ;;   "constant expression required"
 ;; )
+
+;; data segment with memory index 1 (only memory 0 available)
+(assert_invalid
+  (module binary
+    "\00asm" "\01\00\00\00"
+    "\05\03\01"                             ;; memory section
+    "\00\00"                                ;; memory 0
+    "\0b\06\01"                             ;; data section
+    "\01\41\00\0b"                          ;; data segment 0 for memory 1
+    "\00"                                   ;; empty vec(byte)
+  )
+  "unknown memory 1"
+)
+
+;; data segment with memory index 1 (no memory section)
+(assert_invalid
+  (module binary
+    "\00asm" "\01\00\00\00"
+    "\0b\06\01"                             ;; data section
+    "\01\41\00\0b"                          ;; data segment 0 for memory 1
+    "\00"                                   ;; empty vec(byte)
+  )
+  "unknown memory 1"
+)
+
+;; data segment with memory index 1 and malformed index constant expression
+(assert_malformed
+  (module binary
+    "\00asm" "\01\00\00\00"
+    "\0b\03\01"                             ;; data section
+    "\01\00"                                ;; data segment 0 for memory 1 with malformed index expr
+  )
+  "unexpected end of section or function"
+)
```

Comment:
Nit: newline missing

Reply:
I decided to replace this test with an "assert_invalid" one. It detect the same bug in WABT, but contains single error.